pkg/watcher/allocations.go

```diff
@@ -47,22 +49,44 @@ func (w *Watcher) runAllocationWatcher() {
 		log.Debug().Msg("allocations index has changed")
 
 		for _, alloc := range allocations {
-
+			// If the allocation's index hasn't changed, then there is nothing to notify about
 			if !w.indexHasChange(alloc.ModifyIndex, maxFound) {
 				log.Debug().Str("alloc-id", alloc.ID).Msg("allocation index has not changed")
 				continue
 			}
 
-			if len(alloc.TaskGroup) == 0 {
-				continue
-			}
-
+			// If the allocation index *has* changed, check if the change is of interest
 			maxFound = alloc.ModifyIndex
 			log.Info().Str("alloc-id", alloc.ID).Msg("allocation index has changed")
+			if isFiltered(alloc, w.config.nomad.AllocCfg) {
+				continue
+			}
 
 			w.mshChan <- alloc
 		}
 		q.WaitIndex = maxFound
 		w.lastChangeIndex = maxFound
 	}
 }
+
+// isFiltered checks whether a given update about a notification is to be filtered or not
+func isFiltered(alloc *api.AllocationListStub, allocCfg *config.AllocConfig) bool {
+	for _, cs := range allocCfg.ExcludeStates {
```

Comment:
we are discarding the index here so to avoid a copy call
```
for i := range allocCfg.ExcludeStates {
  if alloc.ClientStatus[i] == cs {
```

Reply:
I think you meant this, right? 
```
for i := range allocCfg.ExcludeStates {
    if alloc.ClientStatus == allocCfg.ExcludeStates[i] {
        ...
    }
}
```
If so I've taken your comment on board.